src/runtime_src/core/pcie/tools/xbflash.qspi/xqspips.cpp

```diff
@@ -499,6 +499,24 @@ int XQSPIPS_Flasher::xclReadBack(std::string output, unsigned base, unsigned tot
     unsigned size = 0;
     int beatCount = 0;
 
+    initQSpiPS();
+
+    uint32_t StatusReg = XQSpiPS_GetStatusReg();
+
+    if (StatusReg == 0xFFFFFFFF) {
```

Comment:
This code is dangerous for it makes a lot of assumptions.  For example, what happens of StatusReg becomes a uint64_t.  It would be better to replace:
`0xFFFFFFFF`
with:
`std::numeric_limits<decltype(StatusReg)>::max()`

Reply:
flash spec already determines all registers are 32bit, and the register read/write functions defined are also based on that. and the statusReg variable is also declared as u32

Reply:
and this happens to be 0xffffffff, which means all bits in the register is 1
the value may be a number other than all 1s -- it all depends on how spec defines.
The original code was not written by me -- i just copied it to here. but I don't think using a ::max makes sense here